gapis/gfxapi/gfxapi.proto

```diff
@@ -102,6 +92,26 @@ enum UniformType {
 	Double = 4;
 }
 
+// ResourceData represents the resource state at a single point in a capture
+message ResourceData {
+	oneof data {
+		Texture texture = 1;
+		Shader shader = 2;
+		Program program = 3;
+	}
+}
+
+// Texture represents a texture resource.
+message Texture {
+	oneof type {
```

Comment:
All of the texture types are essentially identical in most respects.
Would you consider changing the "type" to an just an enum value,
and storing the rest in array of layers...
1D, 2D has always just one layer,
3D, 2D_array has arbitrary number of layers,
Cubemap and Cubemap_array have 6*N layers.

Layers->Levels->Data seems to be model used by the Vulkan API, and I have independently switched the GLSL API to follow the same model.
We do not necessarily have to switch in this PR though.